source/AMDSThreadedTCPDataServer.cpp

```diff
@@ -0,0 +1,50 @@
+#include "source/AMDSThreadedTCPDataServer.h"
+
+#include <QCoreApplication>
+
+AMDSThreadedTCPDataServer::AMDSThreadedTCPDataServer(QString hwType, QObject *parent) :
+	QObject(parent)
+{
+	hwType_ = hwType;
+
+	thread_ = new QThread;
+	server_ = new AMDSTCPDataServer;
+
+	connect(thread_, SIGNAL(started()), this, SLOT(onThreadStarted()));
+	connect(this, SIGNAL(startServer(QString,quint16)), server_, SLOT(start(QString,quint16)));
+	connect(server_, SIGNAL(error(quint8,quint16,QString)), this, SIGNAL(error(quint8,quint16,QString)));
+
+	server_->moveToThread(thread_);
+	thread_->start(QThread::LowPriority);
+}
+
+AMDSThreadedTCPDataServer::~AMDSThreadedTCPDataServer()
+{
+	if (thread_->isRunning())
+		thread_->terminate();
+
+	thread_->deleteLater();
+	server_->deleteLater();
+}
+
+AMDSTCPDataServer* AMDSThreadedTCPDataServer::server()
+{
+	return server_;
+}
+
+void AMDSThreadedTCPDataServer::onThreadStarted()
+{
+	QSettings settings;
+	settings.beginGroup("Networking");
+	if(!settings.contains("interface"))
+		settings.setValue("interface", hwType_);
+
+	if(!settings.contains("port"))
+		settings.setValue("port", "28044");
+
+	quint16 port = settings.value("port").toInt();
+	settings.endGroup();
+	qDebug() << "Going to request server to start";
```

Comment:
On further viewing it is obvious there are a bunch of debugs about which could do with being moved over, but I'll assume you're aware of those and, at this stage of development, it isn't an issue.